src/util/std_types.h

```diff
@@ -1048,7 +1048,8 @@ inline const bitvector_typet &to_bitvector_type(const typet &type)
          type.id()==ID_bv ||
          type.id()==ID_pointer ||
          type.id()==ID_c_bit_field ||
-         type.id()==ID_c_bool);
+         type.id()==ID_c_bool ||
+         type.id()==ID_bool);
```

Comment:
bool_typet() is not a bitvector type! In particular, there is no width (this is the key difference between bool and c_bool).